src/zenml/pipelines/run_pipeline.py

```diff
@@ -0,0 +1,222 @@
+        module=module
+    )
+
+    steps = {}
+    for step_name, step_config in config[
+        PipelineConfigurationKeys.STEPS
+    ].items():
+        StepConfigurationKeys.key_check(step_config)
+        source = step_config[StepConfigurationKeys.SOURCE_]
+        step_class = _load_class_from_module(module, source, str(zenml_root))
+
+        # It is necessary to support passing step instances for standard
+        #  step implementations (e.g WhylogsProfilerStep) in order to
+        #  support using the same step multiple  times, once this problem is
+        #  solved, this portion of the code can be simplified to only
+        #  support classes.
+        if not isinstance(step_class, BaseStep):
+            step_instance = step_class()
+        else:
+            step_instance = step_class
+
+        materializers_config = step_config.get(
+            StepConfigurationKeys.MATERIALIZERS_, None
+        )
+        if materializers_config:
+            # We need to differentiate whether it's a single materializer
+            # or a dictionary mapping output names to materializers
+            if isinstance(materializers_config, str):
+                correct_input = textwrap.dedent(
+                    f"""
+                {SourceConfigurationKeys.NAME_}: {materializers_config}
+                {SourceConfigurationKeys.FILE_}: optional/filepath.py
+                """
+                )
+
+                raise PipelineConfigurationError(
+                    "As of ZenML version 0.8.0 `str` entries are no "
+                    "longer supported "
+                    "to define steps or materializers. Instead you will "
+                    "now need to "
+                    "pass a dictionary. This dictionary **has to** "
+                    "contain a "
+                    f"`{SourceConfigurationKeys.NAME_}` which refers to "
+                    f"the function/"
+                    "class name. If this entity is defined outside the "
+                    "main module,"
+                    "you will need to additionally supply a "
+                    f"{SourceConfigurationKeys.FILE_} with the relative "
+                    f"forward-slash-"
+                    "separated path to the file. \n"
+                    f"You tried to pass in `{materializers_config}` "
+                    f"- however you should have specified the name "
+                    f"(and file) like this: \n "
+                    f"{correct_input}"
+                )
+            elif isinstance(materializers_config, dict):
+                materializers = {
+                    output_name: _load_class_from_module(
+                        module, source, str(zenml_root)
+                    )
+                    for output_name, source in materializers_config.items()
+                }
+            else:
+                raise PipelineConfigurationError(
+                    f"Only `str` and `dict` values are allowed for "
```

Comment:
Error above says `str` is also not allowed now. Maybe we can combine those two errors into one that simply checks whether the materializer is a dict?

Reply:
For now I would leave it as two separate messages with the correction of removing str from the second message. Reason being, that if its a str (as it used to be) we can give some additional issue of how to easily change